// data-analysis-python\src\get_holiday_info\get_trading_calendar_jq.proto
syntax = "proto3";

option go_package = "api-go/src/service/ms_gateway/get_trading_calendar_jq.proto";

// GetTradingCalendarJqServiceというサービスを定義
service GetTradingCalendarJqService {
  rpc GetTradingCalendarJq (GetTradingCalendarJqRequest) returns (GetTradingCalendarJqResponse);
}

// リクエストメッセージを定義
message GetTradingCalendarJqRequest {
  string from_date = 1;   // 取得開始日
  string to_date = 2;     // 取得終了日
}

// レスポンスメッセージを定義
message GetTradingCalendarJqResponse {
  repeated TradingCalendar trading_calendar = 1;
}

// 取引カレンダーの詳細を定義
message TradingCalendar {
  string Date = 1;               // 日付
  string HolidayDivision = 2;    // 休日区分
  // 非営業日:0	営業日:1 東証半日立会日:2 非営業日(祝日取引あり):3
}



// python用コンパイルコマンド /data-analysis-pythonにて
// python -m grpc_tools.protoc -I src/get_trading_calendar_jq --python_out src/get_trading_calendar_jq --grpc_python_out=src/get_trading_calendar_jq src/get_trading_calendar_jq/get_trading_calendar_jq.proto

// Go用コンパイルコマンド /api-goにて
// protoc --proto_path=../data-analysis-python/src/get_trading_calendar_jq --go_out=./src/service/ms_gateway/get_trading_calendar_jq --go_opt=paths=source_relative --go-grpc_out=./src/service/ms_gateway/get_trading_calendar_jq --go-grpc_opt=paths=source_relative ../data-analysis-python/src/get_trading_calendar_jq/get_trading_calendar_jq.proto